pkg/readiness/object_tracker.go

```diff
@@ -187,7 +219,7 @@ func (t *objectTracker) Observe(o runtime.Object) {
 		// Satisfy existing expectation
 		delete(t.seen, k)
 		delete(t.expect, k)
-		t.satisfied[k] = struct{}{}
+		t.satisfied[k] = mutatedObjData(t.mutators...)
```

Comment:
We probably want to leave this as an empty struct to avoid unnecessary memory usage.

Reply:
See my comment in other version of this comment.